core/frontend/src/BriefcaseConnection.ts

```diff
@@ -232,7 +232,7 @@ export class BriefcaseConnection extends IModelConnection {
    */
   public async pullAndMergeChanges(version?: IModelVersionProps): Promise<void> {
     this.requireTimeline();
-    return IpcApp.callIpcHost("pullAndMergeChanges", this.key, version);
+    this._changeSetId = await IpcApp.callIpcHost("pullAndMergeChanges", this.key, version);

```

Comment:
Do you think applications might want to be notified via an event when the changesetId changes?

Reply:
`BriefcaseTxns.onChangesPulled` exists but I'm not certain the ordering between that event being dispatched and the IModelConnection's changesetId being updated.

Reply:
the `onChangesPulled` event will happen before the `BriefcaseConnection`'s member is updated.